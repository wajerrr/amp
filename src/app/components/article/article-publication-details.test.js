import React from 'react';
import renderer from 'react-test-renderer';
import ArticlePublicationDetails from './article-publication-details';

describe('ArticlePublicationDetails', () => {
  it('should match snapshot with byline', () => {
    const tree = renderer
      .create(
        <ArticlePublicationDetails
          datePublished="2018-05-08T18:24:40Z"
          byline="me"
        />
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('should match snapshot without byline', () => {
    const tree = renderer
      .create(
        <ArticlePublicationDetails datePublished="2018-05-08T18:24:40Z" />
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
