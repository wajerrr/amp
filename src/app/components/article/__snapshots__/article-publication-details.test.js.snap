// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ArticlePublicationDetails should match snapshot with byline 1`] = `
<div
  className="css-hznrc1-StyledPublicationsContainer escws072"
>
  <div
    className="css-ug5v56-StyledDateAuthor escws074"
  >
    <time
      content="2018-05-08T18:24:40.000Z"
      itemProp="datePublished"
    >
      May 8th 2018
    </time>
     | by 
    <span
      itemProp="author"
    >
      me
    </span>
  </div>
  <div
    className="css-16g3ycg-ShareBarContainer-StyledSocialShare eyrn4830"
  >
    <amp-social-share
      aria-label="Share on Twitter"
      height={31}
      type="twitter"
      width={31}
    >
      <span
        className={undefined}
      >
        <svg
          version="1.1"
          viewBox="0 0 48 48"
          x="0px"
          xmlSpace="preserve"
          xmlns="http://www.w3.org/2000/svg"
          y="0px"
        >
          <path
            d="M33.6,18.9c0,0.3,0,0.4,0,0.7c0,6.6-5,14-14,14c-2.7,0-5.4-0.9-7.6-2.3c0.4,0,0.7,0.1,1.1,0.1 c2.3,0,4.4-0.9,6.1-2.1c-2.1,0-4-1.4-4.6-3.4c0.3,0,0.7,0.1,0.9,0.1c0.4,0,0.9,0,1.3-0.1c-2.3-0.4-4-2.4-4-4.9 c0.7,0.4,1.4,0.7,2.3,0.7c-1.3-0.9-2.3-2.3-2.3-4c0-0.9,0.3-1.7,0.7-2.4c2.4,3,6,5,10.3,5.1c-0.1-0.4-0.1-0.9-0.1-1.1 c0-2.7,2.3-5,5-5c1.4,0,2.7,0.7,3.6,1.6c1.1-0.3,2.3-0.7,3.1-1.1c-0.4,1.1-1.1,2.1-2.3,2.7c1-0.1,2-0.4,2.9-0.9 C35.4,17.4,34.4,18.4,33.6,18.9z"
          />
        </svg>
      </span>
    </amp-social-share>
    <amp-social-share
      aria-label="Share on Facebook"
      data-param-app_id="966242223397117"
      height={31}
      type="facebook"
      width={31}
    >
      <span
        className={undefined}
      >
        <svg
          version="1.1"
          viewBox="0 0 48 48"
          x="0px"
          xmlSpace="preserve"
          xmlns="http://www.w3.org/2000/svg"
          y="0px"
        >
          <path
            d="M29.6,15.2H27c-0.7,0-1.3,0.1-1.4,0.6c-0.1,0.3-0.3,0.7-0.3,1.4v2.1h4.3L29,24h-3.8v13.5h-5.5V24H17v-4.7h2.7 v-2.8c0-2.8,1-4.7,3-5.5c1-0.4,1.7-0.4,2.7-0.4h4.3V15.2z"
          />
        </svg>
      </span>
    </amp-social-share>
    <amp-social-share
      aria-label="Share on LinkedIn"
      height={31}
      type="linkedin"
      width={31}
    >
      <span
        className={undefined}
      >
        <svg
          version="1.1"
          viewBox="0 0 48 48"
          x="0px"
          xmlSpace="preserve"
          xmlns="http://www.w3.org/2000/svg"
          y="0px"
        >
          <path
            d="M17.9,34.5h-4.9V18.6h4.9V34.5z M15.5,16.4c-1.6,0-3-1.3-3-3s1.3-3,3-3c1.6,0,3,1.3,3,3 C18.3,15.2,17,16.4,15.5,16.4z M31.6,34.5V26c0-2.3-0.8-3.5-2.7-3.5c-2,0-3,1.3-3,3.5v8.5h-4.8V18.6h4.8v2.1c0,0,1.4-2.7,4.8-2.7 s5.8,2.1,5.8,6.4s0,10.2,0,10.2H31.6z"
          />
        </svg>
      </span>
    </amp-social-share>
    <amp-social-share
      aria-label="Send an email"
      height={31}
      type="email"
      width={31}
    >
      <span
        className={undefined}
      >
        <svg
          version="1.1"
          viewBox="0 0 48 48"
          x="0px"
          xmlSpace="preserve"
          xmlns="http://www.w3.org/2000/svg"
          y="0px"
        >
          <path
            d="M33.6,14.4H14.4c-1.3,0-2.4,1.1-2.4,2.4l0,14.4c0,1.3,1.1,2.4,2.4,2.4h19.2c1.3,0,2.4-1.1,2.4-2.4V16.8 C36,15.5,34.9,14.4,33.6,14.4z M33.6,19.2l-9.6,6l-9.6-6v-2.4l9.6,6l9.6-6V19.2z"
          />
        </svg>
      </span>
    </amp-social-share>
    
  </div>
</div>
`;

exports[`ArticlePublicationDetails should match snapshot without byline 1`] = `
<div
  className="css-hznrc1-StyledPublicationsContainer escws072"
>
  <div
    className="css-ug5v56-StyledDateAuthor escws074"
  >
    <time
      content="2018-05-08T18:24:40.000Z"
      itemProp="datePublished"
    >
      May 8th 2018
    </time>
    
  </div>
  <div
    className="css-16g3ycg-ShareBarContainer-StyledSocialShare eyrn4830"
  >
    <amp-social-share
      aria-label="Share on Twitter"
      height={31}
      type="twitter"
      width={31}
    >
      <span
        className={undefined}
      >
        <svg
          version="1.1"
          viewBox="0 0 48 48"
          x="0px"
          xmlSpace="preserve"
          xmlns="http://www.w3.org/2000/svg"
          y="0px"
        >
          <path
            d="M33.6,18.9c0,0.3,0,0.4,0,0.7c0,6.6-5,14-14,14c-2.7,0-5.4-0.9-7.6-2.3c0.4,0,0.7,0.1,1.1,0.1 c2.3,0,4.4-0.9,6.1-2.1c-2.1,0-4-1.4-4.6-3.4c0.3,0,0.7,0.1,0.9,0.1c0.4,0,0.9,0,1.3-0.1c-2.3-0.4-4-2.4-4-4.9 c0.7,0.4,1.4,0.7,2.3,0.7c-1.3-0.9-2.3-2.3-2.3-4c0-0.9,0.3-1.7,0.7-2.4c2.4,3,6,5,10.3,5.1c-0.1-0.4-0.1-0.9-0.1-1.1 c0-2.7,2.3-5,5-5c1.4,0,2.7,0.7,3.6,1.6c1.1-0.3,2.3-0.7,3.1-1.1c-0.4,1.1-1.1,2.1-2.3,2.7c1-0.1,2-0.4,2.9-0.9 C35.4,17.4,34.4,18.4,33.6,18.9z"
          />
        </svg>
      </span>
    </amp-social-share>
    <amp-social-share
      aria-label="Share on Facebook"
      data-param-app_id="966242223397117"
      height={31}
      type="facebook"
      width={31}
    >
      <span
        className={undefined}
      >
        <svg
          version="1.1"
          viewBox="0 0 48 48"
          x="0px"
          xmlSpace="preserve"
          xmlns="http://www.w3.org/2000/svg"
          y="0px"
        >
          <path
            d="M29.6,15.2H27c-0.7,0-1.3,0.1-1.4,0.6c-0.1,0.3-0.3,0.7-0.3,1.4v2.1h4.3L29,24h-3.8v13.5h-5.5V24H17v-4.7h2.7 v-2.8c0-2.8,1-4.7,3-5.5c1-0.4,1.7-0.4,2.7-0.4h4.3V15.2z"
          />
        </svg>
      </span>
    </amp-social-share>
    <amp-social-share
      aria-label="Share on LinkedIn"
      height={31}
      type="linkedin"
      width={31}
    >
      <span
        className={undefined}
      >
        <svg
          version="1.1"
          viewBox="0 0 48 48"
          x="0px"
          xmlSpace="preserve"
          xmlns="http://www.w3.org/2000/svg"
          y="0px"
        >
          <path
            d="M17.9,34.5h-4.9V18.6h4.9V34.5z M15.5,16.4c-1.6,0-3-1.3-3-3s1.3-3,3-3c1.6,0,3,1.3,3,3 C18.3,15.2,17,16.4,15.5,16.4z M31.6,34.5V26c0-2.3-0.8-3.5-2.7-3.5c-2,0-3,1.3-3,3.5v8.5h-4.8V18.6h4.8v2.1c0,0,1.4-2.7,4.8-2.7 s5.8,2.1,5.8,6.4s0,10.2,0,10.2H31.6z"
          />
        </svg>
      </span>
    </amp-social-share>
    <amp-social-share
      aria-label="Send an email"
      height={31}
      type="email"
      width={31}
    >
      <span
        className={undefined}
      >
        <svg
          version="1.1"
          viewBox="0 0 48 48"
          x="0px"
          xmlSpace="preserve"
          xmlns="http://www.w3.org/2000/svg"
          y="0px"
        >
          <path
            d="M33.6,14.4H14.4c-1.3,0-2.4,1.1-2.4,2.4l0,14.4c0,1.3,1.1,2.4,2.4,2.4h19.2c1.3,0,2.4-1.1,2.4-2.4V16.8 C36,15.5,34.9,14.4,33.6,14.4z M33.6,19.2l-9.6,6l-9.6-6v-2.4l9.6,6l9.6-6V19.2z"
          />
        </svg>
      </span>
    </amp-social-share>
    
  </div>
</div>
`;
